Jenkinsfile-utils.groovy

```diff
@@ -0,0 +1,109 @@
+/*
+noisePageBuild will create a build directory and compile and build the
+noisepage binary in that directory. The options passed into the method
+determine the compilation and build options. Refer to the defaultArgs
+for the different options that can be passed in.
+*/
+def noisePageBuild(Map args = [:]){
+    def defaultArgs = [
+        useCache: true,
+        buildType: "Debug",
+        os: "ubuntu",
+        isBuildTests: true,
+        useASAN: false,
+        isBuildBenchmarks: false,
+        isCodeCoverage: false,
+        isJumboTest: false,
+        isRecordTime: false
+    ]
+    def Map config = defaultArgs << args
+    def String compileCmd = generateCompileCmd(config)
+    def String buildCmd = generateBuildCmd(config)
+    def String buildScript = generateBuildScript(compileCmd, buildCmd, config.isRecordTime)
+
+    sh script:buildScript, label: "Compile & Build"
+}
+
+/*
+generateCompileCmd creates the cmake command string. It is based on the
+config passed into the function. The config arguments are the same as the
+defaultArgs in noisePageBuild.
+*/
```

Comment:
Add leading `*` prefix to function comments to match rest of the system:

```
/**
 * generateCompileCmd creates the cmake command string. It is based on the
 * config passed into the function. The config arguments are the same as the
 * defaultArgs in noisePageBuild.
 */
```

Example: https://github.com/cmu-db/noisepage/blob/master/src/include/main/db_main.h#L52-L56